src/stuf/stuf_bg/tests/test_client.py

```diff
@@ -37,98 +34,73 @@ def _stuf_bg_xmlschema_doc():
+            },
+        ),
+        encoding="utf-8",
+    )
+    # assert this mocked response is valid
+    xmlschema = etree.XMLSchema(_stuf_bg_xmlschema_doc())
+    xmlschema.assert_(_extract_soap_body(response_body))
+    return response_body
 
-        self.assertEqual(m.last_request.method, "POST")
-        self.assertEqual(
-            m.last_request.headers["Content-Type"],
-            SOAP_VERSION_CONTENT_TYPES.get(SOAPVersion.soap11),
-        )
-        self.assertEqual(
-            m.last_request.headers["SOAPAction"],
-            "http://www.egem.nl/StUF/sector/bg/0310/npsLv01",
-        )
 
-        with open(
-            f"{settings.BASE_DIR}/src/stuf/stuf_bg/xsd/bg0310/vraagAntwoord/bg0310_namespace.xsd",
-            "r",
-        ) as f:
-            xmlschema_doc = etree.parse(f)
-            xmlschema = etree.XMLSchema(xmlschema_doc)
-
-            doc = etree.parse(BytesIO(m.last_request.body))
-            el = (
-                doc.getroot()
-                .xpath(
-                    "soap:Body",
-                    namespaces={"soap": "http://schemas.xmlsoap.org/soap/envelope/"},
-                )[0]
-                .getchildren()[0]
-            )
-            if not xmlschema.validate(el):
-                self.fail(
-                    f'Request body "{m.last_request.body}" is not valid against StUF-BG XSDs. '
-                    f"Error: {xmlschema.error_log.last_error.message}"
-                )
-        self.assertEqual(
-            TimelineLogProxy.objects.filter(
-                template="logging/events/stuf_bg_request.txt"
-            ).count(),
-            1,
-        )
-        self.assertEqual(
-            TimelineLogProxy.objects.filter(
-                template="logging/events/stuf_bg_request.txt"
-            ).count(),
-            1,
-        )
+class StufBGConfigTests(TestCase):
+    def test_client_requires_a_service(self):
+        config = StufBGConfig.get_solo()
 
-    def test_all_attributes_are_mapped(self):
+        with self.assertRaises(RuntimeError):
+            config.get_client()
+
+    def test_all_prefill_attributes_are_mapped(self):
         available_attributes = FieldChoices.values
         for attribute in available_attributes:
             with self.subTest(attribute=attribute):
                 glom_target = ATTRIBUTES_TO_STUF_BG_MAPPING.get(attribute)
-                if not glom_target:
-                    self.fail(f"unmapped attribute: {attribute}")
+                self.assert_(glom_target, f"unmapped attribute: {attribute}")
+
+
+class StufBGClientTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        cls.config = StufBGConfig.get_solo()
+        cls.config.service = StufServiceFactory.create(soap_service__soap_version="1.1")
+        cls.config.save()
+
+    def setUp(self):
+        super().setUp()
+        self.client = self.config.get_client()
```

Comment:
looking at the `StufBGClient` class itself, I don't see any use of `StufBGConfig` in there. It seems to me that we can bypass that entirely?

```py
def setUp(self):
    service = StufServiceFactory.build(soap_service__soap_version="1.1")
    self.client = StufBGClient(service)
```

combining that with a separate test that asserts that `StufBGConfig().get_client()` returns a client instance seems sufficient to me (you already have a test for the misconfiguration case)